jwt/jwt.go

```diff
@@ -20,16 +20,27 @@ package jwt
 import (
 	"gopkg.in/square/go-jose.v2"
 	"gopkg.in/square/go-jose.v2/json"
+	"strings"
 )
 
 // JSONWebToken represents a JSON Web Token (as specified in RFC7519).
 type JSONWebToken struct {
+	err     error
```

Comment:
So this means that I could have a `JSONWebToken` struct, but the struct doesn't actually represent a valid token. That seems like it might be confusing. Is there a reason it has to be this way?